stdlib/public/core/Integers.swift.gyb

```diff
@@ -3232,257 +3183,496 @@ public struct DoubleWidth<T : FixedWidthInteger>
+
+    let integerPart: Raw = bitPattern &>> Raw(offset)
 
+    // Exit early for zero
+    guard integerPart > 0 else {
+      self.init(0)
+      return
+    }
+
+    if source.sign == .minus {
+      if !DoubleWidth.isSigned || integerPart &- 1 > DoubleWidth.max {
+        return nil
+      }
+      self.init(integerPart &- 1)
+      self = 0 &- self &- 1
+    } else {
+      self.init(exactly: integerPart)
+    }
+  }
+    
   public func _word(at n: Int) -> UInt {
-    if T.bitWidth < ${word_bits} || T.bitWidth % ${word_bits} != 0 {
-      fatalError("_word(at:) is not supported on this type")
+    if Base.bitWidth < ${word_bits} {
+      if ${word_bits} % Base.bitWidth != 0 {
+        fatalError("word(at:) is not supported on this type")
+      }
+      if n > 0 {
+        return _storage.high._word(at: n)
+      }
+      return _storage.low._word(at: 0) | _storage.high._word(at: 0) << numericCast(Base.bitWidth)
+    } else {
+      // multiples of ${word_bits} only
+      if Base.bitWidth % ${word_bits} != 0 {
+        fatalError("word(at:) is not supported on this type")
+      }
+
+      // TODO: move to Int128 just like init(_builtinIntegerLiteral:) ?
+      let wordsInBase = Base.bitWidth / ${word_bits}
+      return (n < wordsInBase) ?
+        _storage.low._word(at: n) :
+        _storage.high._word(at: n - wordsInBase)
     }
-    // TODO: move to Int128 just like init(_builtinIntegerLiteral:) ?
-    let wordsInT = T.bitWidth / ${word_bits}
-    return (n < wordsInT) ?
-      _storage.low._word(at: n) :
-      _storage.high._word(at: n - wordsInT)
   }
 
   public static var isSigned: Bool {
-    return T.isSigned
+    return Base.isSigned
   }
 
   // fixed width
   //
-  public static var max: DoubleWidth<T> {
+  public static var max: DoubleWidth {
     return self.init((High.max, Low.max))
   }
 
-  public static var min: DoubleWidth<T> {
+  public static var min: DoubleWidth {
     return self.init((High.min, Low.min))
   }
 
-  public static var bitWidth : Int { return 2 * T.bitWidth }
+  public static var bitWidth: Int {
+    return 2 * Base.bitWidth
+  }
 
 % # This covers + and -
 % for x in binaryArithmetic['Numeric'][:2]:
-%   highAffectedByLowOverflow = 'T.max' if x.operator == '+' else 'T.min'
-  public func ${x.name}ReportingOverflow(_ rhs: DoubleWidth<T>)
-    -> (partialValue: DoubleWidth<T>, overflow: ArithmeticOverflow) {
+%   highAffectedByLowOverflow = 'Base.max' if x.operator == '+' else 'Base.min'
+  public func ${x.name}ReportingOverflow(_ rhs: DoubleWidth)
+    -> (partialValue: DoubleWidth, overflow: ArithmeticOverflow) {
     let (low, lowOverflow) =
       _storage.low.${x.name}ReportingOverflow(rhs._storage.low)
     let (high, highOverflow) =
       _storage.high.${x.name}ReportingOverflow(rhs._storage.high)
     let isLowOverflow = lowOverflow == .overflow
-    let result = (high ${x.operator} (isLowOverflow ? 1 : 0), low)
+    let result = (high &${x.operator} (isLowOverflow ? High(1) : High(0)), low)
```

Comment:
`High` conforms to `ExpressibleByIntegerLiteral`, and you removed the explicit usage of types in favor of just literals elsewhere =)